backbone.js

```diff
@@ -1408,6 +1408,31 @@
     return Backbone.$.ajax.apply(Backbone.$, arguments);
   };
 
+  // A psuedo Promise implementation used to ensure asynchronous methods
+  // return thenables.
+  // Override this if you'd like to use a different ES6 library.
+  Backbone.Promise = function() {
```

Comment:
How about a `if (typeof Promise != "undefined") Backbone.Promise = Promise`
line right after our pollyfill using deferreds? You should probably be
pollyfilling Promise anyways if you need it.

On Sun, May 31, 2015, 21:23 Tim Griesser <notifications@github.com> wrote:

> In backbone.js
> <https://github.com/jashkenas/backbone/pull/3651#discussion_r31396998>:
>
> > @@ -1408,6 +1408,31 @@
> >      return Backbone.$.ajax.apply(Backbone.$, arguments);
> >    };
> >
> > +  // A psuedo Promise implementation used to ensure asynchronous methods
> > +  // return thenables.
> > +  // Override this if you'd like to use a different ES6 library.
> > +  Backbone.Promise = function() {
>
>  We could go our "jQuery require" route, attempting to require some
> library 'promise', then feed that into the factory function. Or just
> require overwriting BB.Promise on node if you want to use async methods.
> Hmmm.
>
> As of ES6, we can just expect Promise it to be a builtin global (and it
> is already in many environments <http://caniuse.com/#search=promise>) so
> it'd be exactly as Brad describes, the same as polyfilling JSON. The only
> requirement here would be that BB wouldn't depend on any specific
> non-standard library specific methods/helpers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/backbone/pull/3651/files#r31396998>.
>